MantidQt/CustomInterfaces/inc/MantidQtCustomInterfaces/Tomography/TomographyProcess.h

```diff
@@ -0,0 +1,87 @@
+#ifndef MANTIDQTCUSTOMINTERFACES_TOMOGRAPHY_TOMOGRAPHYPROCESS_H_
+#define MANTIDQTCUSTOMINTERFACES_TOMOGRAPHY_TOMOGRAPHYPROCESS_H_
+
+#include <QProcess>
+#include <QString>
+#include <QStringList>
+
+namespace MantidQt {
+namespace CustomInterfaces {
+
+/*
+TomographyProcess class that run external processes and provides some helper
+functions. This class was designed to be used with TomographyThread to run
+external processes asyncronously to the main thread of Mantid.
+
+Mantid is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 3 of the License, or
+(at your option) any later version.
+
+Mantid is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+File change history is stored at: <https://github.com/mantidproject/mantid>
+Code Documentation is available at: <http://doxygen.mantidproject.org>
+*/
+class TomographyProcess : public QProcess {
+  Q_OBJECT
+public:
+  // we want a nullptr for parent so we can move it to a thread
+  TomographyProcess() : QProcess(nullptr) {}
+
+  // intentionally copy the vector
+  void setup(const std::string runnable, const std::vector<std::string> &args,
+             const std::string &allOpts) {
+    m_allArgs = allOpts;
+    m_runnable = std::move(QString::fromStdString(runnable));
+    m_args = std::move(constructArgumentsFromVector(args));
```

Comment:
Same with `std::move` here - the compiler can do RVO if it thinks its appropriate